test/jdk/javax/swing/SwingWorker/TestDoneBeforeDoInBackground.java

```diff
@@ -0,0 +1,147 @@
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+/*
+ * @test
+ * @bug 8081474
+ * @summary  Verifies if SwingWorker calls 'done'
+ *           before the 'doInBackground' is finished
+ * @run main TestDoneBeforeDoInBackground
+ */
+import java.beans.PropertyChangeListener;
+import java.beans.PropertyChangeEvent;
+import javax.swing.SwingWorker;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class TestDoneBeforeDoInBackground {
+
+    private static final int WAIT_TIME = 200;
+    private static final long CLEANUP_TIME = 1000;
+
+    private static final AtomicBoolean doInBackgroundStarted = new AtomicBoolean(false);
+    private static final AtomicBoolean doInBackgroundFinished = new AtomicBoolean(false);
+    private static final AtomicBoolean doneFinished = new AtomicBoolean(false);
+    private static final CountDownLatch doneLatch = new CountDownLatch(1);
+
+    public static void main(String[] args) throws InterruptedException {
+        SwingWorker<String, String> worker = new SwingWorker<>() {
+            @Override
+            protected String doInBackground() throws Exception {
+                try {
+                    while (!Thread.currentThread().isInterrupted()) {
+                        System.out.println("Working...");
+                        Thread.sleep(WAIT_TIME);
+                    }
+                } catch (InterruptedException ex) {
+                    System.out.println("Got interrupted!");
+                }
+
+                try {
+                    doInBackgroundStarted.set(true);
+                    System.out.println("Cleaning up");
+                    Thread.sleep(CLEANUP_TIME);
+                    System.out.println("Done cleaning");
+                    doInBackgroundFinished.set(true);
+                } catch (InterruptedException ex) {
+                    System.out.println("Got interrupted second time!");
+                }
+
+                return null;
+            }
+
+            @Override
+            protected void done() {
+                if (!doInBackgroundFinished.get()) {
+                    throw new RuntimeException("done called before " +
+                                               "doInBackground is finished");
+                }
+                System.out.println("Done");
+                doneFinished.set(true);
+                doneLatch.countDown();
+            }
+        };
+
+        worker.addPropertyChangeListener(
+            new PropertyChangeListener() {
+                public void propertyChange(PropertyChangeEvent evt) {
+                    System.out.println("doInBackground started: " +
+                                        doInBackgroundStarted.get() +
```

Comment:
```suggestion
                    System.out.println("doInBackgroundStarted: " +
                                        doInBackgroundStarted.get() +
```
Just for consistency with the following `-Finished` line.